# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CacheIRCompiler.cpp
# Commit: 6042c5c7c2a0
# Full Hash: 6042c5c7c2a02d1fdc08354b801ed6b89da67bd3
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2020-05-07 21:43:47
# Regressor Bug: 1635878
# File Overlap Count: 1
# Description:
#   Bug 1635878 - CacheIR: Optimize TypedArray length. r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D74138
# ==============================================================================

diff -r 762b51f271f4 -r 6042c5c7c2a0 js/src/jit/CacheIRCompiler.cpp
--- a/js/src/jit/CacheIRCompiler.cpp	Thu May 07 18:02:49 2020 +0000
+++ b/js/src/jit/CacheIRCompiler.cpp	Thu May 07 12:21:38 2020 +0000
@@ -3097,6 +3097,17 @@
   return true;
 }
 
+bool CacheIRCompiler::emitLoadTypedArrayLengthResult(ObjOperandId objId) {
+  JitSpew(JitSpew_Codegen, "%s", __FUNCTION__);
+  AutoOutputRegister output(*this);
+  Register obj = allocator.useRegister(masm, objId);
+  AutoScratchRegisterMaybeOutput scratch(allocator, masm, output);
+
+  masm.unboxInt32(Address(obj, TypedArrayObject::lengthOffset()), scratch);
+  EmitStoreResult(masm, scratch, JSVAL_TYPE_INT32, output);
+  return true;
+}
+
 bool CacheIRCompiler::emitLoadFunctionLengthResult(ObjOperandId objId) {
   JitSpew(JitSpew_Codegen, "%s", __FUNCTION__);
   AutoOutputRegister output(*this);